bootstrap_cfn/autoscale.py

```diff
@@ -46,3 +53,145 @@ def get_all_autoscaling_groups(self):
             response = self.conn_asg.get_all_groups(next_token=response.next_token)
             all_asgs += response
         return all_asgs
+
+    def cycle_instances(self,
+                        termination_delay=None):
+        """
+        Cycle all the instances in an autoscaling group, waiting for the
+        specified delay before terminating each instance that was replaced
+
+        Args:
+            termination_delay(int): The delay in seconds between the new instance becoming
+                healthy and in-service, and the termination of the old one its replacing.
+        """
+        client = boto3.client('autoscaling')
+
+        # Use the type of health check the ASG is using to determine a sensible default for termination
+        # delay. The ELB check is more nuanced and should know what a healthy service really looks like.
+        # EC2 checks are basic and generally, the instance will be up and 'healthy' long before the service
+        # is available. In that case we want to delay termination for a long enough time to hope the service
+        # sets itself up.
+        if not termination_delay and self.group.health_check_type != "ELB":
+            termination_delay = 360
+
+        # Get a list of the current instances
+        current_instance_ids = [instance.get('InstanceId') for instance in self.get_healthy_instances()]
+        logging.getLogger("bootstrap-cfn").info("cycle_instances: Found {} instance ids, {}"
+                                                .format(len(current_instance_ids), current_instance_ids))
+
```

Comment:
Is it worth presenting a status of all instances at this stage (not just the healthy ones)? Assuming someone will be changing their states it might be worth giving out a quick check like: 
`Healthy: 3  Unhealthy: 0`  <-- this would at least make you feel better that there were 0 unhealthy
Or something like `Healthy:  i-whatever1, i-whatever2    Other: i-whatever3`  <-- even better, it shows you which was not healthy/terminating/changing to begin with.

This would help us track "what happened" in case for example one of the instances was not healthy to begin with.